docs/source/tutorials_source/package/tutorial_simsiam_esa.py

```diff
@@ -0,0 +1,443 @@
+# create a lightly dataset for embedding
+dataset_test = lightly.data.LightlyDataset(
+    input_dir=path_to_data,
+    transform=test_transforms
+)
+
+
+
+# create a dataloader for embedding
+dataloader_test = torch.utils.data.DataLoader(
+    dataset_test,
+    batch_size=batch_size,
+    shuffle=False,
+    drop_last=False,
+    num_workers=num_workers
+)
+
+# %%
+# Create the SimSiam model
+# -----------------------
+#
+# Create a ResNet backbone and remove the classification head
+
+# we use a pretrained resnet for this tutorial to speed
+# up training time but you can also train one from scratch
+resnet = torchvision.models.resnet18()
+backbone = nn.Sequential(*list(resnet.children())[:-1])
+
+# create the SimSiam model using the backbone from above
+model = lightly.models.SimSiam(
+    backbone,
+    num_ftrs=num_ftrs,
+    proj_hidden_dim=pred_hidden_dim,
+    pred_hidden_dim=pred_hidden_dim,
+    out_dim=out_dim,
+    num_mlp_layers=num_mlp_layers
+)
+
+# %%
+# SimSiam uses a symmetric negative cosine similarity loss and does therefore
+# not require any negative samples. We build a criterion and an optimizer.
+
+# SimSiam uses a symmetric negative cosine similarity loss
+criterion = lightly.loss.SymNegCosineSimilarityLoss()
+
+# scale the learning rate 
+lr = 0.05 * batch_size / 256
+# use SGD with momentum and weight decay
+optimizer = torch.optim.SGD(
+    model.parameters(),
+    lr=lr,
+    momentum=0.9,
+    weight_decay=5e-4
+)
+
+
+# %%
+# Train SimSiam
+# --------------------
+# 
+# To train the SimSiam model, you can use a classic PyTorch training loop:
+# For every epoch, iterate over all batches in the training data, extract
+# the two transforms of every image, pass them through the model, and calculate
+# the loss. Then, simply update the weights with the optimizer. Don't forget to
+# reset the gradients!
+#
+# Since SimSiam doesn't require negative samples, it is a good idea to check 
+# whether the outputs of the model have collapsed into a single direction. For
+# this we can simply check the standard deviation of the L2 normalized output
+# vectors. If it is close to one divided by the square root of the output 
+# dimension, everything is fine (you can read
+# up on this idea `here <https://arxiv.org/abs/2011.10566>`_).
+
+device = 'cuda' if torch.cuda.is_available() else 'cpu'
+model.to(device)
+
+avg_loss = 0.
+avg_output_std = 0.
+for e in range(1, epochs + 1):
+
+    for (x0, x1), _, _ in dataloader_train_simsiam:
+
+        # move images to the gpu
+        x0 = x0.to(device)
+        x1 = x1.to(device)
+
+        # run the model on both transforms of the images
+        y0, y1 = model(x0, x1)
+
+        # backpropagation
+        loss = criterion(y0, y1)
+        loss.backward()
+
+        optimizer.step()
+        optimizer.zero_grad()
+
+        # calculate the per-dimension standard deviation of the outputs
+        # we can use this later to check whether the embeddings are collapsing
+        output, _ = y0
```

Comment:
I miss somehow the information that the SimSiam forward pass returns a tuple of (prediction, projection). Maybe adding a comment or renaming the variables would help?